components/brave_new_tab_ui/components/default/cryptoDotCom/preOptInView.tsx

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// you can obtain one at http://mozilla.org/MPL/2.0/.
+
+import * as React from 'react'
+
+import { currencyNames } from './data'
+import {
+  ActionButton,
+  Box,
+  FlexItem,
+  PlainButton,
+  Text
+} from './style'
+
+import {
+  formattedNum,
+  getPercentColor,
+  renderIconAsset,
+  transformLosersGainers
+} from './utils'
+
+import { getLocale } from '../../../../common/locale'
+interface Props {
+  optInBTCPrice: boolean
+  losersGainers: Record<string, chrome.cryptoDotCom.AssetRanking[]>
+  tickerPrices: Record<string, chrome.cryptoDotCom.TickerPrice>
+  onBTCPriceOptedIn: () => void
+  getCryptoDotComClientUrl: (callback: (clientAuthUrl: string) => void) => void
+}
+
+export default function PreOptInView ({
+  optInBTCPrice,
+  tickerPrices,
+  losersGainers,
+  onBTCPriceOptedIn,
+  getCryptoDotComClientUrl
+}: Props) {
+  const currency = 'BTC'
+  const { price = null } = tickerPrices[`${currency}_USDT`] || {}
+  const transformedLosersGainers = transformLosersGainers(losersGainers || {})
+  const { percentChange = null } = transformedLosersGainers[currency] || {}
+
+  const [clientAuthUrl, setClientAuthUrl] = React.useState('')
+
+  const onClickConnectToCryptoDotCom = () => {
+    window.open(clientAuthUrl, '_self', 'noopener')
+  }
+
+  getCryptoDotComClientUrl((clientAuthUrl: string) => {
+    setClientAuthUrl(clientAuthUrl)
+  })
+
+  return (
+    <>
+      <Box isFlex={true} $height={48}>
+        <FlexItem $pr={5}>
+          {renderIconAsset(currency.toLowerCase())}
+        </FlexItem>
+        <FlexItem>
+            <Text>{currency}</Text>
+            <Text small={true} textColor='light'>{currencyNames[currency]}</Text>
+        </FlexItem>
+        <FlexItem textAlign='right' flex={1}>
+          {optInBTCPrice ? (
+            <>
+              {(price !== null) && <Text>{formattedNum(price)}</Text>}
```

Comment:
price could be null before getting any responce from service.
and price will be `$0.00` after getting invalid responce from crypto.com service. In this case, it's set by `CryptoDotComService::OnTickerInfo()`.
Otherwise, valid valud will be passed.

Reply:
Should we change it so that it's `null` after getting invalid response? Otherwise user may think valid value is `$0.00`, or will be confused.

Reply:
Fixed by https://github.com/brave/brave-core/pull/7699/commits/a46e423d0ff1bcdfc1ed8041b0d4484f0757fab0